lib/js-yaml/dumper.js

```diff
@@ -738,7 +739,7 @@ function writeNode(state, level, object, block, compact, iskey) {
       if (block && (0 !== Object.keys(state.dump).length)) {
         writeBlockMapping(state, level, state.dump, compact);
         if (duplicate) {
-          state.dump = '&ref_' + duplicateIndex + (0 === level ? '\n' : '') + state.dump;
+          state.dump = '&ref_' + duplicateIndex + state.dump;
```

Comment:
Yes, I removed it completely. I will check if your example still works correctly. In the meantime my reasoning to remove this condition:

The problem with the incorrectly prepended YAML alias names only occurred if `compact` was true. `writeNode` is called with `compact = true` from lines [536](https://github.com/nodeca/js-yaml/blob/0215138ec880c1e0ebc93f5a8c94bb753f5a5456/lib/js-yaml/dumper.js#L536) and from line [625](https://github.com/nodeca/js-yaml/blob/0215138ec880c1e0ebc93f5a8c94bb753f5a5456/lib/js-yaml/dumper.js#L625) and line [830](https://github.com/nodeca/js-yaml/blob/0215138ec880c1e0ebc93f5a8c94bb753f5a5456/lib/js-yaml/dumper.js#L830).

`(0 === level ? '\n' : '')` fixes prepending the alias for top level objects, i.e. for calls from line 830. For example, this snippet ensures that we get

```yaml
&ref_0
- *ref_0
```

instead of

```yaml
&ref_0- *ref_0
```

The two other positions which called `writeNode` with `compact = true` were not handled correctly by this condition.

Since I set `compact` to `false` when an alias is prepended, I think that this condition is not necessary anymore. Without the removal of `(0 === level ? '\n' : '')` we would now actually get

```yaml
&ref_0

- *ref_0
```

(note the extra empty line)